src/tools/migrations/public/index.ts

```diff
@@ -0,0 +1,67 @@
+import 'tsconfig-paths/register'
+import 'dotenv/config'
+
+import * as fs from 'fs'
+import * as path from 'path'
+
+import { DB } from 'server/db'
+import { Logger } from 'server/utils/logger'
+
+const client = DB
+let migrationSteps: Array<string>
+let previousMigrations: Array<string> = []
+const executedMigrations: Array<string> = []
+
+// TODO: Move to migration step / schema initialisation
+const tableDDL = `
+    create table if not exists public.migrations (
+      id integer not null default nextval('public.migrations_id_seq'::regclass),
+      name character varying(255) primary key not null,
+      run_on timestamp without time zone not null default now()
+    );
+`
+
+const init = async () => {
+  await client.query(tableDDL)
+  previousMigrations = await client.map('select * from migrations', [], (row) => row.name)
+  migrationSteps = fs
+    .readdirSync(path.join(__dirname, `steps`))
+    .filter((file) => file !== 'template.ts' && file.endsWith('.ts') && !previousMigrations.includes(file))
+}
+
+const close = async () => {
+  // Note: Omitted REDIS related code. Add them here if needed.
+  await DB.$pool.end()
+}
```

Comment:
for now i omit the redis related code, eg. connection closing etc as i don't think we need them in DDL manipulation
can be added as needed